drivers/include/net/netdev2.h

```diff
@@ -10,15 +10,153 @@
  */
 
 /**
- * @defgroup    drivers_netdev_netdev2 Generic Low-Level Network Driver Interface v2
+ * @defgroup    drivers_netdev_netdev2 Network Device Driver Interface
  * @ingroup     drivers_netdev
  * @brief       This is the second version of a generic low-level network
  *              driver interface
  * @{
  *
- * This interface is supposed to be a low-level interface for network drivers.
+ * @section sec_about About
  *
- * Example call flow:
+ * This interface provides a uniform API for network stacks to interact with
+ * network device drivers. This interface is designed in a way, that it is
+ * completely agnostic to the used network stack. This way, device drivers for
+ * network devices (e.g. IEEE802.15.4 radios, Ethernet devices, ...) have to
+ * implemented once and can be used with any (future) supported network stack
+ * in RIOT.
+ *
+ * The functions provided by the interface cover three major parts: (i) sending
+ * and receiving of actual network data, (ii) network device configuration
+ * through reading and setting device parameters, and (iii) event handling.
+ *
+ *
+ * @section sec_problem The Interrupt Context Problem
```

Comment:
I have the feeling that this whole section should come later. Someone who is writing a new device driver doesn't necessarily want to know about the details of interrupt contexts first, but first *how* to write a network device driver. Maybe mention somewhere why we do things like we do (=> because of the interrupt context problem) and refer to this section (which then comes later).